src/spikeinterface/extractors/cellexplorersortingextractor.py

```diff
@@ -1,136 +1,163 @@
-    Spike times are stored in units of seconds.
+    Spike times are stored in units of seconds so we transform them to units of samples.
+
+    The newer version of the format is described here:
+    https://cellexplorer.org/data-structure/
+
+    Whereas the old format is described here:
+    https://github.com/buzsakilab/buzcode/wiki/Data-Formatting-Standards
 
     Parameters
     ----------
-    spikes_matfile_path : PathType
-        Path to the sorting_id.spikes.cellinfo.mat file.
+    folder_path: str | Path
+        Path to the folder containing the .mat files from the session.
+    sampling_frequency: float | None, optional
+        The sampling frequency of the data. If None, it will be extracted from the files.
+    session_info_matfile_path: str | Path | None, optional
+        Path to the session info .mat file. If None, it will be inferred from the folder path.
     """
 
     extractor_name = "CellExplorerSortingExtractor"
-    installed = HAVE_SCIPY_AND_HDF5STORAGE
     is_writable = True
-    mode = "file"
-    installation_mesg = "To use the CellExplorerSortingExtractor install scipy and hdf5storage: \n\n pip install scipy  hdf5storage"
+    mode = "folder"
+    installation_mesg = "To use the CellExplorerSortingExtractor install scipy and h5py"
+
+    def __init__(
+        self,
+        folder_path: str | Path,
+        sampling_frequency: float | None = None,
+        session_info_matfile_path: str | Path | None = None,
+    ):
+        try:
+            import h5py
+            import scipy.io
+        except ImportError:
+            raise ImportError(self.installation_mesg)
 
-    def __init__(self, spikes_matfile_path: PathType,
-                 session_info_matfile_path: OptionalPathType=None,
-                 sampling_frequency: Optional[float] = None):
-        assert self.installed, self.installation_mesg
+        self.folder_path = Path(folder_path)
+        self.session_info_matfile_path = session_info_matfile_path
+        
+        mat_file_path = next((path for path in self.folder_path.iterdir() if path.suffix == ".mat"), None) 
+        self.session_id = mat_file_path.stem.split(".")[0]
+        assert self.session_id is not None, f"No .mat files found in the {self.folder_path}!"
 
-        spikes_matfile_path = Path(spikes_matfile_path)
-        assert (
-            spikes_matfile_path.is_file()
-        ), f"The spikes_matfile_path ({spikes_matfile_path}) must exist!"
+        spikes_cellinfo_path = self.folder_path / f"{self.session_id}.spikes.cellinfo.mat"
+        assert spikes_cellinfo_path.is_file(), f"The spikes.cellinfo.mat file must exist in the {self.folder_path}!"
         
+        read_as_hdf5 = False 
+        try:
+            spikes_mat = scipy.io.loadmat(file_name=str(spikes_cellinfo_path), simplify_cells=True)
+        except NotImplementedError:
+            spikes_mat = h5py.File(name=spikes_cellinfo_path, mode="r")
+            read_as_hdf5 = True
```

Comment:
A fair bit cleaner this way; there may be some minor syntax changes between `hdf5storage` and `h5py` when accessing array fields however

Reply:
Hoping the 2 cell explorer datasets on GIN are representative of what you might see with those differences

Reply:
Yep, I think it is a little bit cleaner. To be honest, when I started doing this earlier this afternoon I had the hope that I would be able to keep the spikes acess lazy for the case when matlab new format is used (something that hdf5 storage does not do). It was a little bit more complicated than I thought. I might come back to this if I have more time when we are doing the full conversion with all the sessions.

Reply:
We have three datasets that cover all the cases for the previous format (sessionInfo in new vs old format and spikes.cellinfo in new vs old format). 

This is thanks to you btw, @CodyCBakerPhD . You made me do it back in the day : P and then you went and reduce the file to upload it to gin.

I also have a file from the repo with the new format that I tested locally.

Reply:
I also added the third datasets to the tests.